### ISSUE TITLE
[FEATURE] Vault Transit support

### ISSUE THREAD

**Issue by danlsgiga** [hashicorp/nomad#5171]
It would be really nice to have Nomad supporting Vault Transit encryption so we can manage secrets directly in git and Nomad would only decrypt it using the Transit endpoint at runtime using the ciphertext in the `template` stanza.

It would also provide a performance boost on the Vault servers for static secrets since Nomad won't need to watch for changes.

IMO, it would be a game changing feature!

**Comment by dadgar** [comment]
@danlsgiga Should this be instead a feature request against consul-template?

**Comment by danlsgiga** [comment]
@dadgar indeed! Only after creating it I realized the `template` stanza is basically a `consul-template` integration. Will request this feature there!

Thanks for the heads up!

**Comment by danlsgiga** [comment]
@dadgar indeed! Only after creating it I realized the `template` stanza is basically a `consul-template` integration. Will request this feature there!

Thanks for the heads up!
